kale/workflow_objects.py

```diff
@@ -366,10 +396,36 @@ def draw_dag(self, layout=None):
         toolbar = bq.Toolbar(figure=fig)
 
         return ipw.VBox([fig, toolbar])
+    
+    def check_selection_for_jumps(self):
+        """
+        It's okay for a selcted task to have a parent or child
+        which is not selected. But it's not okay to have
+        a generational gap in selection.
+        """
+        pass
 
     def gen_subdag(self):
         """Return DAG containing only steps which are to be run. (Not yet implemented.)"""
-        return self.dag
+        subdag = networkx.DiGraph()
+
+        if self._bqgraph.selected is None or len(self._bqgraph.selected) == 0:
+            return self.dag
+        else:
+            self.check_selection_for_jumps()
+
+            for node in self.dag.nodes():
+                node_index = node.index[self]
+                if node_index in self._bqgraph.selected:
+                    subdag.add_node(node)
+                    parent_list = list(self.dag.predecessors(node))
+                    if len(parent_list) > 0:
```

Comment:
this if is not needed, the for loop will not execute if parent_list has 0 elements